src/MOIWrapper.jl

```diff
@@ -0,0 +1,397 @@
+                S == MOI.ZeroOne && push!(zeroOneIndices, mapping.varmap[f.variable].value)
+                S == MOI.Integer && push!(integerIndices, mapping.varmap[f.variable].value)
+            end
+        end
+
+        ## Update conmap for (F,S) constraints
+        F == MOI.SingleVariable && continue
+        for i in 1:length(ci)
+            mapping.conmap[ci[i]] = MOI.ConstraintIndex{F,S}(nbRows + i)
+        end
+        nbRows += MOI.get(userOptimizer, MOI.NumberOfConstraints{F,S}())
+
+    end
+
+    cbcModelFormat = CbcModelFormat(nbRows, nbCols)
+
+    for (F,S) in listOfConstraints
+        (F,S) in INTEGRALITY_CONSTRAINTS && continue
+        for ci in MOI.get(userOptimizer, MOI.ListOfConstraintIndices{F,S}())
+            f = MOI.get(userOptimizer, MOI.ConstraintFunction(), ci)
+            s = MOI.get(userOptimizer,  MOI.ConstraintSet(), ci)
+            loadConstraint(ci, cbcModelFormat, mapping, f, s)
+        end
+    end
+
+    ## Add bounds to binary columns if it makes sense
+    for colIdx in zeroOneIndices
+        if cbcModelFormat.col_lb[colIdx] < 0.0
+            cbcModelFormat.col_lb[colIdx] = 0.0
+        end
+        if cbcModelFormat.col_ub[colIdx] > 1.0
+            cbcModelFormat.col_ub[colIdx] = 1.0
+        end
+    end
+
+
+    ## Copy objective function
+    objF = MOI.get(userOptimizer, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}())
+    loadObj(cbcModelFormat, mapping, objF)
+    cbcOptimizer.objFunctConstant = objF.constant
+    sense = MOI.get(userOptimizer, MOI.ObjectiveSense())
+    MOI.set!(cbcOptimizer, MOI.ObjectiveSense(), sense)
+
+    @show cbcModelFormat.constraint_matrix
+    @show cbcModelFormat.col_lb
+    @show cbcModelFormat.col_ub
+    @show cbcModelFormat.row_lb
+    @show cbcModelFormat.row_ub
+    @show cbcModelFormat.obj
+    @show CbcCI.getObjSense(cbcOptimizer.inner)
+
+    ## Load the problem to Cbc
+    CbcCI.loadProblem(cbcOptimizer.inner, cbcModelFormat.constraint_matrix, cbcModelFormat.col_lb, cbcModelFormat.col_ub, cbcModelFormat.obj, cbcModelFormat.row_lb, cbcModelFormat.row_ub)
+
+    ## Set integer variables
+    for colIdx in vcat(integerIndices, zeroOneIndices)
+        CbcCI.setInteger(cbcOptimizer.inner, colIdx-1)
+    end
+
+    return MOI.CopyResult(MOI.CopySuccess, "Model was copied succefully.", MOIU.IndexMap(mapping.varmap, mapping.conmap))
+end
+
+
+function MOI.optimize!(cbcOptimizer::CbcOptimizer)
+    # Call solve function
+    CbcCI.solve(cbcOptimizer.inner)
+end
+
+
+
+## canadd, canset, canget functions
+
+function MOI.canaddvariable(cbcOptimizer::CbcOptimizer)
+    return false
+end
+
+function MOI.cantransformconstraint(model::CbcOptimizer, c::MOI.ConstraintIndex{F}, ::Type{S}) where {F<:MOI.AbstractFunction, S<:MOI.AbstractSet}
+    return false
+end
+
+## supports constraints
+
+function MOI.supportsconstraint(cbcOptimizer::CbcOptimizer, F::Type{<:MOI.AbstractFunction}, S::Type{<:MOI.AbstractScalarSet})
```

Comment:
It would be more efficient to do check if it is supported at compile time by doing
```julia
MOI.supportsconstraint(::CbcOptimizer, ::Type{<:Union{...}}, ::Type{<:Union{...}})
```
and it would also make the result of `supportsconstraint` inferrable so calls to `supportsconstraint` can be optimized out by the compiler.